src/server/ua_services_nodemanagement.c

```diff
@@ -477,6 +489,18 @@ copyChild(UA_Server *server, UA_Session *session, const UA_NodeId *destinationNo
         UA_NodeId_deleteMembers(&node->nodeId);
         node->nodeId.namespaceIndex = destinationNodeId->namespaceIndex;
 
+        if (server->config.nodeLifecycle.generateChildNodeId) {
+            retval = server->config.nodeLifecycle.generateChildNodeId(server,
+                                                                      &session->sessionId, session->sessionHandle,
+                                                                      &rd->nodeId.nodeId,
+                                                                      destinationNodeId,
+                                                                      &rd->referenceTypeId,
+                                                                      &node->nodeId);
+            if(retval != UA_STATUSCODE_GOOD)
+                return retval;
```

Comment:
Now you have to delete the node, if you return here.
Otherwise we lose memory!

Reply:
Oops. Sorry, I'll have a look.